kernel/kernel-api/src/test/scala/io/delta/kernel/internal/SnapshotManagerSuite.scala

```diff
@@ -775,31 +776,43 @@ class SnapshotManagerSuite extends AnyFunSuite with MockFileSystemClientUtils {
 }
 
 trait SidecarIteratorProvider extends VectorTestUtils {
-  def singletonSidecarIterator(sidecars: Seq[FileStatus])
-      : CloseableIterator[ColumnarBatch] = Utils.singletonCloseableIterator(
-    new ColumnarBatch {
-      override def getSchema: StructType = SidecarFile.READ_SCHEMA
-
-      override def getColumnVector(ordinal: Int): ColumnVector = {
-        ordinal match {
-          case 0 => stringVector(sidecars.map(_.getPath)) // path
-          case 1 => longVector(sidecars.map(_.getSize): _*) // size
-          case 2 =>
-            longVector(sidecars.map(_.getModificationTime): _*); // modification time
-        }
-      }
 
-      override def getSize: Int = sidecars.length
-    })
+  // Shared helper to construct the batch
+  private def buildSidecarBatch(sidecars: Seq[FileStatus]): ColumnarBatch = new ColumnarBatch {
+    override def getSchema: StructType = SidecarFile.READ_SCHEMA
+
+    override def getColumnVector(ordinal: Int): ColumnVector = ordinal match {
+      case 0 => stringVector(sidecars.map(_.getPath)) // path
+      case 1 => longVector(sidecars.map(_.getSize): _*) // size
+      case 2 => longVector(sidecars.map(_.getModificationTime): _*) // modification time
+    }
+
+    override def getSize: Int = sidecars.length
+  }
+
+  // For Parquet (ColumnarBatch + filePath)
```

Comment:
IMO this comment and thte JSON comment are superfluous -- if/when we refactor the JSON code, there is a chance we forget to refactor that JSON comment. we can delete this comment

btw -- method comments should use the form `/** my comment */